var/spack/repos/builtin/packages/neurodamus-core/package.py

```diff
@@ -45,6 +50,12 @@ def install(self, spec, prefix):
             copy_all('resources/common/hoc', prefix.hoc)
             copy_all('resources/common/mod', prefix.mod)
 
-    def setup_environment(self, spack_env, run_env):
+        filter_file(r'UNKNOWN_CORE_VERSION', r'%s' % spec.version, prefix.hoc.join('defvar.hoc'))
+        git_fetcher = self.get_git_fetcher()
+        if git_fetcher is not None:
```

Comment:
Instead of all these indirections, you could just ask for forgiveness:
```python
try:
    commit_hash = self.fetcher[0].get_commit()
    filter_file(r'UNKNOWN_CORE_HASH', r'\'{}\''.format(commit), prefix.hoc.join('defvar.hoc'))
except:
    pass
```
IMO that's a lot cleaner, and generic enough that the underlying RCS does not matter.

Reply:
I think that should be done for both files. Do we care that it's a `Git<Whatever>Fetcher`? All we care about is whether or not it provides us with a `commit`, or even better, `version` thing. You could catch an `AttributeError` to be on the safer side and re-raise if it's not complaining about `get_commit`, but I think this short version conveys everything needed and would serve our purposi.